cmd/subnetcmd/add_validator.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package subnetcmd
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/prompts"
+	"github.com/ava-labs/avalanche-cli/pkg/subnet"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/spf13/cobra"
+)
+
+var (
+	nodeIDStr    string
+	weightStr    string
+	startTimeStr string
+	duration     time.Duration
+
+	errNoSubnetID = errors.New("failed to find the subnet ID for this subnet, has it been deployed/created?")
+)
+
+// avalanche subnet deploy
+func newAddValidatorCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "addValidator [subnetName]",
+		Short: "Allow a validator to stake on the given subnet",
+		Long: `The addValidator command prompts for start time, duration and weight for validating this subnet.
+It also prompts for the NodeID of the node which will be validating this subnet.`,
+		SilenceUsage: true,
+		RunE:         addValidator,
+		Args:         cobra.ExactArgs(1),
+	}
+	cmd.Flags().StringVarP(&keyName, "key", "k", "", "select the key to use")
+	cmd.Flags().StringVar(&nodeIDStr, "nodeID", "", "set the NodeID of the validator to add")
+	cmd.Flags().StringVar(&weightStr, "weight", "", "set the staking weight of the validator to add")
+	cmd.Flags().StringVar(&startTimeStr, "start-time", "", "start time when this validator starts validating, in 'YYYY-MM-DD HH:MM:SS' format")
+	cmd.Flags().DurationVar(&duration, "staking-period", 0, "how long this validator will be staking")
+	return cmd
+}
+
+func addValidator(cmd *cobra.Command, args []string) error {
+	var (
+		nodeID ids.NodeID
+		weight uint64
+		start  time.Time
+		err    error
+	)
+
+	if keyName == "" {
```

Comment:
Let's prompt if the name is empty

Reply:
Ah no, mistery solved, it was/is in the deploy command...